compiler/noirc_frontend/src/tests.rs

```diff
@@ -2136,3 +2136,157 @@ fn no_super() {
     assert_eq!(span.start(), 4);
     assert_eq!(span.end(), 9);
 }
+
+#[test]
+fn cannot_call_unconstrained_function_outside_of_unsafe() {
+    let src = r#"
+    fn main() {
+        foo();
+    }
+
+    unconstrained fn foo() {}
+    "#;
+    let errors = get_program_errors(src);
+    assert_eq!(errors.len(), 1);
+
+    let CompilationError::TypeError(TypeCheckError::Unsafe { .. }) = &errors[0].0 else {
+        panic!("Expected an 'unsafe' error, got {:?}", errors[0].0);
+    };
+}
+
+#[test]
+fn cannot_call_unconstrained_captured_function_outside_of_unsafe() {
```

Comment:
```suggestion
fn cannot_call_unconstrained_first_class_function_outside_of_unsafe() {
```
terminology: The function isn't captured in a closure, it's just used as a first class value